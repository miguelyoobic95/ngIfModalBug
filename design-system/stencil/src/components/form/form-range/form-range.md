---
name: Range
category: Form Components
---
