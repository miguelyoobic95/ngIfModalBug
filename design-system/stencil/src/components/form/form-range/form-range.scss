
:host {
    .inputs-container {
        align-items: center;
        padding: 0 0.3rem;
        .separator {
            margin: 0 0.875rem;
            width: 0.75rem;
            height: 0.125rem;
            border-radius: 1px;
            background: cssvar(stable);
        }
        .input {
            height: 2.5rem;
            border-radius: 5px;
            border: 1px solid cssvar(stable);
            display: flex;
            align-items: center;
            padding-left: 1rem;
            &.single {
                width: 100%;
            }
            input[type=number]{
                -webkit-appearance: none;
                background: transparent;
                font-size: 0.875rem;
                width: 100%;
                //width: 2rem;
                outline: none;
                border-color: transparent;
                //text-align: center;
            }
            input[type=number]::-webkit-inner-spin-button,
            input[type=number]::-webkit-outer-spin-button{
                -webkit-appearance: none;
            }
            // Firefox:
            input[type=number] {
                -moz-appearance:textfield;
              }
        }
    }
    .slider-container {
        display: flex;
        align-items: center;
        justify-content: center;
        yoo-form-slider {
            width: 100%;
            /deep/ .wrapper {
                width: 100%;
                .outer-container {
                    width: 100%;                    
                    .slider-container {
                        width: 100%;                            
                        .range-container {
                            width: 100%;
                            input {
                                width: 100% !important;
                            }
                        }
                    }
                }
            }
        }
    }
}