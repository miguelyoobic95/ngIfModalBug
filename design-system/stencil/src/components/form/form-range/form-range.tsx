import { Component, Prop, Event, State, EventEmitter, Element, Method } from '@stencil/core';
import { ValidatorEntry, AsyncValidator, Validator, IFormRange, IRangeValue } from '@shared/interfaces';
import { setValidator, setAsyncValidator, setValueAndValidateInput } from '../../../utils/helpers/form-input-helpers';
// import { setValidator, setAsyncValidator, onInputBlurred, onInputChanged, onInputFocused, onFocus } from '../../../utils/helpers/form-input-helpers';

@Component({
    tag: 'yoo-form-range',
    styleUrl: 'form-range.scss',
    scoped: true
})
export class YooFormRangeComponent implements IFormRange {

    @Prop({ mutable: true }) value: IRangeValue;
    @Prop() validators: Array<Validator<IRangeValue> | ValidatorEntry> = [];
    @Prop() asyncValidators: Array<AsyncValidator<IRangeValue>>;
    @Prop() readonly: boolean;
    @Prop() min: number;
    @Prop() max: number;
    @Prop() double: boolean; // if simple we use value.sup

    @Event() validityChanged: EventEmitter<boolean>;
    @Event() inputBlurred: EventEmitter<any>;
    @Event() inputFocused: EventEmitter<boolean>;
    @Event() inputChanged: EventEmitter<any>;

    @Event() iconClicked: EventEmitter<string>;

    @State() validity: boolean;

     @Element() host: HTMLStencilElement;

    // private coreConfig: ICoreConfig = (window as any).coreConfigService;

    // Reduced Validators
    _validator: Validator<string> = (x: string) => true;
    _asyncValidator: AsyncValidator<string> = async (x: string) => true;

    componentWillLoad() {
        setValidator(this.validators);
        setAsyncValidator(this.asyncValidators);
    }

    componentDidLoad() {
        this.value = {...this.value};
    }

    @Method()
    isValid() {
        return this.validity;
    }

    onInputClear(): void {
        this.iconClicked.emit('clear');
    }

    onSliderChange(ev: CustomEvent): void {
        let oldValue: IRangeValue = this.value;
        if (ev && ev.detail) {
            if (ev.detail.lowValue || ev.detail.lowValue === 0) {
                this.value.inf = ev.detail.lowValue;
            }
            if (ev.detail.highValue || ev.detail.highValue === 0) {
                this.value.sup = ev.detail.highValue;
            }
        }
        if (oldValue !== this.value) {
            setValueAndValidateInput({target: {value: this.value}}, this);
        }
        this.value = {...this.value};
    }

    onSingleSliderChange(ev: CustomEvent): void {
        let oldValue: IRangeValue = this.value;
        if (ev && ev.detail) {
            this.value.sup = ev.detail;
        }
        if (oldValue !== this.value) {
            setValueAndValidateInput(this.value, this);
        }
        this.value = {...this.value};
    }

    onInputInfChanged(ev): void {
        let oldValue: IRangeValue = this.value;
        this.value.inf = this.value.sup >= ev.target.valueAsNumber ? ev.target.valueAsNumber : this.value.sup;
        this.value.sup = this.value.sup >= ev.target.valueAsNumber ? this.value.sup : ev.target.valueAsNumber;
        if (oldValue !== this.value) {
            setValueAndValidateInput([this.value.inf, this.value.sup], this);
        }
        this.value = {...this.value};
    }

    onInputSupChanged(ev): void {
        let oldValue: IRangeValue = this.value;
        if (this.double ) {
            this.value.sup = this.value.inf <= ev.target.valueAsNumber ? ev.target.valueAsNumber : this.value.inf;
            this.value.inf = this.value.inf <= ev.target.valueAsNumber ? this.value.inf : ev.target.valueAsNumber;
        } else {
            this.value.sup = ev.target.valueAsNumber;
        }
        if (oldValue !== this.value) {
            setValueAndValidateInput([this.value.inf, this.value.sup], this);
        }
        this.value = {...this.value};
    }

    renderReadonly() {
        return (
            <div class="readonly">
                <div>
                {this.value ? this.value.inf : null}
                </div>
                <div>
                {this.value ? this.value.inf : null}
                </div>
            </div>
        );
    }

    renderEditable() {
        return (
            <div class="outer-container" attr-layout="column">
                <div class="inputs-container" attr-layout="row">
                    {this.double ?
                         [<div class="input">
                            <input type="number" value={this.value && this.value.inf ? this.value.inf.toString() || null : null} onChange={(ev) => this.onInputInfChanged(ev)}/>
                        </div>,
                        <div class="separator"></div>]
                    : null}
                    <div class={'input ' + (this.double ? '' : 'single')}>
                        <input type="number" value={this.value && this.value.sup ? this.value.sup || null : null} onChange={(ev) => this.onInputSupChanged(ev)}/>
                    </div>
                </div>
                <div class="slider-container">
                    <yoo-form-slider class="gradient-success"
                        hideLabel={true}
                        hideReferences={true}
                        doubleSlider={this.double}
                        initialLowValue={this.value ? this.value.inf : 0}
                        initialValue={this.value ? this.value.sup : 0}
                        minimum={this.min ? this.min : null}
                        maximum={this.max ? this.max : null}
                        onDoubleSliderChanged={(ev) => this.onSliderChange(ev)}
                        onSingleSliderChanged={(ev) => this.onSingleSliderChange(ev)}
                        ></yoo-form-slider>
                </div>
            </div>
        );
    }

    render(): JSX.Element {
        return this.readonly ? this.renderReadonly() : this.renderEditable();
    }

}
