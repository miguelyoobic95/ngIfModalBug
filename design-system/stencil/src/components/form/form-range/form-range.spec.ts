import { TestWindow } from '@stencil/core/testing';
import { YooFormRangeComponent } from './form-range';

describe('YooFormInputComponent', () => {
    let window;
    beforeEach(() => {
      window = new TestWindow();
    });
    it('should build', () => {
        expect(new YooFormRangeComponent()).toMatchSnapshot();
    });

    describe('rendering', () => {
        beforeEach( async () => {
        });
        it ('should render simple text input', async () => {
            let element = await window.load({
                components: [YooFormRangeComponent],
                html: '<yoo-form-range></yoo-form-range>'
            });
            expect(element).toMatchSnapshot();
        });

        it ('should render simple text input', async () => {
            let element = await window.load({
                components: [YooFormRangeComponent],
                html: '<yoo-form-range></yoo-form-range>'
            });
            element.value = {inf: 11, sup: 38};
            element.min = 0;
            element.max = 100;
            element.double = true;
            expect(element).toMatchSnapshot();
        });

        it ('should render simple text input', async () => {
            let element = await window.load({
                components: [YooFormRangeComponent],
                html: '<yoo-form-range></yoo-form-range>'
            });
            element.value = {inf: 11, sup: 38};
            element.min = 0;
            element.max = 100;
            element.double = false;
            expect(element).toMatchSnapshot();
        });
    });

});

