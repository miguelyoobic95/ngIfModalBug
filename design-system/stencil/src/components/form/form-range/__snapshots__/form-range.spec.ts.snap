// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YooFormInputComponent rendering should render simple text input 1`] = `
<yoo-form-range
  class="hydrated"
  data-yoo-form-range-host=""
>
  
  <div
    attr-layout="column"
    class="outer-container"
    data-yoo-form-range=""
  >
    <div
      attr-layout="row"
      class="inputs-container"
      data-yoo-form-range=""
    >
      
      <div
        class="input single"
        data-yoo-form-range=""
      >
        <input
          data-yoo-form-range=""
          type="number"
        />
      </div>
    </div>
    <div
      class="slider-container"
      data-yoo-form-range=""
    >
      <yoo-form-slider
        class="gradient-success"
        data-yoo-form-range=""
        hidelabel="true"
        hidereferences="true"
        initiallowvalue="0"
        initialvalue="0"
      />
    </div>
  </div>
</yoo-form-range>
`;

exports[`YooFormInputComponent rendering should render simple text input 2`] = `
<yoo-form-range
  class="hydrated"
  data-yoo-form-range-host=""
>
  
  <div
    attr-layout="column"
    class="outer-container"
    data-yoo-form-range=""
  >
    <div
      attr-layout="row"
      class="inputs-container"
      data-yoo-form-range=""
    >
      
      <div
        class="input single"
        data-yoo-form-range=""
      >
        <input
          data-yoo-form-range=""
          type="number"
        />
      </div>
    </div>
    <div
      class="slider-container"
      data-yoo-form-range=""
    >
      <yoo-form-slider
        class="gradient-success"
        data-yoo-form-range=""
        hidelabel="true"
        hidereferences="true"
        initiallowvalue="0"
        initialvalue="0"
      />
    </div>
  </div>
</yoo-form-range>
`;

exports[`YooFormInputComponent rendering should render simple text input 3`] = `
<yoo-form-range
  class="hydrated"
  data-yoo-form-range-host=""
>
  
  <div
    attr-layout="column"
    class="outer-container"
    data-yoo-form-range=""
  >
    <div
      attr-layout="row"
      class="inputs-container"
      data-yoo-form-range=""
    >
      
      <div
        class="input single"
        data-yoo-form-range=""
      >
        <input
          data-yoo-form-range=""
          type="number"
        />
      </div>
    </div>
    <div
      class="slider-container"
      data-yoo-form-range=""
    >
      <yoo-form-slider
        class="gradient-success"
        data-yoo-form-range=""
        hidelabel="true"
        hidereferences="true"
        initiallowvalue="0"
        initialvalue="0"
      />
    </div>
  </div>
</yoo-form-range>
`;

exports[`YooFormInputComponent should build 1`] = `
YooFormRangeComponent {
  "_asyncValidator": [Function],
  "_validator": [Function],
  "validators": Array [],
}
`;
